index.js

```diff
@@ -136,8 +136,9 @@ function renderTag(elem, opts) {
   }
 
   if (
-    opts.xmlMode
-    && (!elem.children || elem.children.length === 0)
+    (opts.xmlMode
+    && (!elem.children || elem.children.length === 0)) ||
```

Comment:
These are to be separate checks set `tag += '/>'` when
`opts.xmlMode   && (!elem.children || elem.children.length === 0)` matches or when `singleTag[elem.name] && opts.recognizeSelfClosing`